mne/time_frequency/tests/test_spectrum.py

```diff
@@ -295,44 +432,135 @@ def _check_spectrum_equivalent(spect1, spect2, tmp_path):
     assert_array_equal(spect1.freqs, spect2.freqs)
 
 
-def test_spectrum_array_errors(epochs_spectrum):
-    """Test EpochsSpectrumArray constructor errors."""
-    data, freqs = epochs_spectrum.get_data(return_freqs=True)
-    info = epochs_spectrum.info
-    with pytest.raises(ValueError, match="Data must be a 3D array"):
-        EpochsSpectrumArray(np.empty((2, 3, 4, 5)), info, freqs)
+@pytest.mark.parametrize("kind", ("raw", "epochs"))
+@pytest.mark.parametrize(
+    "method, output, average",
+    [
+        ("welch", "power", "mean"),  # test with precomputed spectrum
+        ("welch", "power", False),  # unaggregated segments
+        ("multitaper", "complex", None),  # unaggregated tapers
+    ],
+)
+def test_spectrum_array_errors(kind, method, output, average, request):
+    """Test (Epochs)SpectrumArray constructor errors."""
```

Comment:
Here just extending to account for the new possible errors, e.g. passing data with segments/tapers to `Array` classes without specifying method.
Also testing now with raw data which wasn't there before, but there aren't any raw-specific errors that need to be checked.

Reply:
don't bother testing with Raw (*in this test*) if there aren't any Raw-specific errors (because it probably isn't improving coverage, and because our test suite takes too long already)